### PR TITLE
GRAPHICS: Enable blitting 24bpp sources in managed_surface.cpp

### PR DESCRIPTION
**Pull request by greg-kennedy**
Function `blitFromInner()` excludes source formats with 3 bytes per pixel.  These are commonly returned from opaque full-color image decoders, like JPEG.  This change enables blitting with these source images.

- Add `== 3` to the list of acceptable source formats
- Add READ_UINT24 handling for 3-byte source formats
- Add READ_UINT24 and RGBA conversion for 3-byte destination when blending (copying non-opaque source to 24bpp surface)

### CODE REVIEW

File: graphics/managed_surface.cpp [hunk 1]
```diff
@@ -386,18 +387,15 @@ void ManagedSurface::blitFromInner(const Surface &src, const Common::Rect &srcRe
...
-					} else {
-						aDest = 0xFF;
-						rDest = destVal[0];
-						gDest = destVal[1];
-						bDest = destVal[2];
-					}
+					uint32 destColor;
+					if (destFormat.bytesPerPixel == 2)
+						destColor = *reinterpret_cast<uint16 *>(destVal);
+					else if (format.bytesPerPixel == 4)
```
Comment (OMGPizzaGuy) [inline graphics/managed_surface.cpp:393]: Oh, while we're here - this reference to format probably should be the local destFormat like the previous check

Comment (greg-kennedy) [reply graphics/managed_surface.cpp]: copy-paste error again, fixed in amended commit

File: graphics/managed_surface.cpp [hunk 2]
```diff
@@ -386,18 +387,15 @@ void ManagedSurface::blitFromInner(const Surface &src, const Common::Rect &srcRe
...
-						uint32 destColor = *reinterpret_cast<uint16 *>(destVal);
-						destFormat.colorToARGB(destColor, aDest, rDest, gDest, bDest);
-					} else if (format.bytesPerPixel == 4) {
-						uint32 destColor = *reinterpret_cast<uint32 *>(destVal);
-						destFormat.colorToARGB(destColor, aDest, rDest, gDest, bDest);
-					} else {
-						aDest = 0xFF;
-						rDest = destVal[0];
-						gDest = destVal[1];
-						bDest = destVal[2];
```
Comment (bluegr) [inline graphics/managed_surface.cpp:399]: Why did you remove this case? This triggers when we have a 1bpp format

Comment (lephilousophe) [reply graphics/managed_surface.cpp:399]: `destFormat.bytesPerPixel` is not allowed to be 1 as per the assert in the beginning of the function.

Comment (lephilousophe) [reply graphics/managed_surface.cpp:399]: Ah sorry, I spoke too fast. It's allowed when formats are the same.

Comment (lephilousophe) [reply graphics/managed_surface.cpp:399]: But it's handled on line 339

Comment (bluegr) [reply graphics/managed_surface.cpp:399]: You are right, thanks @lephilousophe

**Comment by bluegr** [comment]
Nice work, thanks!
